jwst/source_catalog/source_catalog.py

```diff
@@ -1,222 +1,1312 @@
-
-    skew_tolerance : float, optional
-        The absolute allowable difference between the x and y axis
-        rotations.
+    model : `ImageModel`
+        The input `ImageModel`.
 
     Returns
     -------
-    rot : float
-        The counterclockwise rotation angle of the image.
+    total_error : `~numpy.ndarray`
+        The total error array.
+    """
+
+    # TODO: Until errors are produced for the level 3 drizzle
+    # products, the JPWG has decided that the errors should not be
+    # populated.
+    return np.zeros_like(model.data)
+
+
+class SourceCatalog:
     """
+    Class for the source catalog.
+
+    Parameters
+    ----------
+    model : `ImageModel`
+        The input `ImageModel`.  The data is assumed to be
+        background-subtracted.
+
+    segment_image : `~photutils.segmentation.SegmentationImage`
+        A 2D segmentation image, with the same shape as the input data,
+        where sources are marked by different positive integer values.
+        A value of zero is reserved for the background.
+
+    error : array_like or `~astropy.units.Quantity`, optional
+        The total error array corresponding to the input ``data`` array.
+        ``error`` is assumed to include *all* sources of error,
+        including the Poisson error of the sources (see
+        `~photutils.utils.calc_total_error`) .  ``error`` must have the
+        same shape as the input ``data``.
+
+    kernel : array-like (2D) or `~astropy.convolution.Kernel2D`, optional
+        The 2D array of the kernel used to filter the data prior to
+        calculating the source centroid and morphological parameters.
+        The kernel should be the same one used in defining the source
+        segments.
+
+    kernel_fwhm : float
+        The full-width at half-maximum (FWHM) of the 2D Gaussian kernel.
+
+    aperture_params : `dict`
+        A dictionary containing the aperture parameters (radii, aperture
+        corrections, and background annulus inner and outer radii).
+
+    abvega_offset : float
+        Offset to convert from AB to Vega magnitudes.  The value
+        reprents m_AB - m_Vega.
+    """
+
+    def __init__(self, model, segment_img, error=None, kernel=None,
+                 kernel_fwhm=None, aperture_params=None, abvega_offset=0.0):
+
+        if not isinstance(model, ImageModel):
+            raise ValueError('The input model must be a ImageModel.')
+        self.model = model  # background was subtracted in SourceDetection
+
+        self.segment_img = segment_img
+        self.error = error  # total error array
+        self.kernel = kernel
+        self.kernel_fwhm = kernel_fwhm
+        self.aperture_params = aperture_params
+        self.abvega_offset = abvega_offset
+
+        self.aperture_ee = aperture_params['aperture_ee']
+        self.n_aper = len(self.aperture_ee)
+        self.column_desc = {}
+
+        # self.wcs = self.model.meta.wcs  # gWCS
+        self.wcs = self.model.get_fits_wcs()  # FITS WCS
```

Comment:
I discovered a bug (https://github.com/spacetelescope/gwcs/issues/294) preventing me from using `gwcs`.  This can be changed to use `gwcs` when that is fixed.